provider/core/src/marker.rs

```diff
@@ -108,6 +108,45 @@ impl<M: DataMarker + Sized> DataMarkerExt for M {
     }
 }
 
+/// This trait should be implemented on all data structs.
+///
+/// For most data structs, a default implementation available via
+/// [`does_not_deref_to_varule`] can be used.
+pub trait MaybeAsVarULE {
```

Comment:
thought: not the best fan of this name, would _kind of_ prefer this talk about the optimizaiton it signals rather than whether or not the type can be converted to a VarULE.

We will have many types that can be VarULEd but I'm not convinced we should automatically apply the optimization there.

Reply:
Classic "what is is versus what it does" naming problem. Happy for naming suggestions. Here are a few:

- `StoreAsVarULE`
- `MaybeStoreAsVarULE`
- `OptimizeForVarZeroVec`
- `WithEfficientBakeStorage`

Reply:
MaybeStoreAsVarULE is fine, or even better MaybeExportAsVarULE. WithEfficientBakeStorage isn't better because we may also use this in postcard.

Reply:
Changed it to `MaybeExportAsVarULE`